examples/camera-controller/webrtc_manager/WebRTCProviderClient.cpp

```diff
@@ -70,6 +85,12 @@ CHIP_ERROR WebRTCProviderClient::ProvideICECandidates(uint16_t webRTCSessionID,
 {
     ChipLogProgress(Camera, "Sending ProvideICECandidates to node " ChipLogFormatX64, ChipLogValueX64(mPeerId.GetNodeId()));
 
+    if (mState != State::Idle)
```

Comment:
Recommend having a state like `HaveRemoteAnswer` at the point when it is sending the ICE candidates. This can only be sent after the initial Offer/Answer flow has completed.

Reply:
1. If we have already sent ICE candidates in ProvideOffer, we will not send ProvideICECandidates
2. If non normal flow, the controller won't receive the remote answer.

Shall we use 'AwaitingICECandidates' to cover all of those cases?

Reply:
It seems it is very tricky to use state to handle ICECandidates exchange, right now, we just use FSM to guard SDP exchange, we can add additional state to handle ICECandidates exchange later

Reply:
Ok. Let's file an issue and link to keep track of this and revisit later if we need to expand the states of the FSM.

Reply:
https://github.com/project-chip/connectedhomeip/issues/38350